File: app/src/main/java/com/example/testproject/network/interceptor/ProgressInterceptor.java
package com.example.testproject.network.interceptor;

import android.support.annotation.NonNull;
import okhttp3.Interceptor;
import okhttp3.Request;
import okhttp3.Response;

import java.io.IOException;

/**
 * Created by mike.li.
 * 进度
 */

public class ProgressInterceptor implements Interceptor {
    @Override
    public Response intercept(@NonNull Chain chain) throws IOException {
        Request request = chain.request();
        Response response = chain.proceed(request);
        return response.newBuilder().body(new ProgressResponseBody(response.body())).build();
    }
}

File: app/src/main/java/com/example/testproject/network/interceptor/ProgressInterceptor.kt
package com.example.testproject.network.interceptor

import okhttp3.Interceptor
import okhttp3.Request
import okhttp3.Response

import java.io.IOException

/**
 * Created by mike.li.
 * 进度
 */

class ProgressInterceptor : Interceptor {
    @Throws(IOException::class)
    override fun intercept(chain: Interceptor.Chain): Response {
        val request = chain.request()
        val response = chain.proceed(request)
        return response.newBuilder().body(response.body()?.let { ProgressResponseBody(it) }).build()
    }
}
